src/types/line.js

```diff
@@ -1,36 +1,59 @@
 // Get the chart variable
-var Chart = require('chart.js');
-Chart = typeof(Chart) === 'function' ? Chart : window.Chart;
-var chartHelpers = Chart.helpers;
 var helpers = require('../helpers.js');
 
 // Line Annotation implementation
 module.exports = function(Chart) {
+	var chartHelpers = Chart.helpers;
+
 	var horizontalKeyword = 'horizontal';
 	var verticalKeyword = 'vertical';
 
-	var LineAnnotation = Chart.Element.extend({
-		setRanges: function(options) {
+	var LineAnnotation = Chart.Annotation.Element.extend({
+		inRange: function(mouseX, mouseY) {
+			var model = this._model || {};
+			return (model.line && model.line.intersects(mouseX, mouseY, this.getHeight()));
+		},
+		getCenterPoint: function() {
+			return {
+				x: (this._view.x2 + this._view.x1) / 2,
+				y: (this._view.y2 + this._view.y1) / 2
+			};
+		},
+		getWidth: function() {
+			return Math.abs(this._view.right - this._view.left);
+		},
+		getHeight: function() {
+			return this._view.borderWidth || 1;
+		},
+		getArea: function() {
+			return Math.sqrt(Math.pow(this.getWidth(), 2) + Math.pow(this.getHeight(), 2));
+		},
+		setRanges: function() {
 			var model = this._model = chartHelpers.clone(this._model) || {};
 
+			// Set the data range for this annotation
 			model.ranges = {};
-			model.ranges[options.scaleID] = {
-				min: options.value,
-				max: options.endValue || options.value
+			model.ranges[this.options.scaleID] = {
+				min: this.options.value,
+				max: this.options.endValue || this.options.value
 			};
 		},
-		configure: function(options, chartInstance) {
+		configure: function() {
 			var model = this._model = chartHelpers.clone(this._model) || {};
 
-			var scale = chartInstance.scales[options.scaleID];
+			var scale = this.chartInstance.scales[this.options.scaleID];
 			var pixel, endPixel;
 			if (scale) {
-				pixel = helpers.isValid(options.value) ? scale.getPixelForValue(options.value) : NaN;
-				endPixel = helpers.isValid(options.endValue) ? scale.getPixelForValue(options.endValue) : pixel;
+				pixel = helpers.isValid(this.options.value) ? scale.getPixelForValue(this.options.value) : NaN;
+				endPixel = helpers.isValid(this.options.endValue) ? scale.getPixelForValue(this.options.endValue) : pixel;
+			}
+
+			if (isNaN(pixel)) {
+				return;
 			}
 
-			var chartArea = chartInstance.chartArea;
-			var ctx = chartInstance.chart.ctx;
+			var chartArea = this.chartInstance.chartArea;
```

Comment:
[minor] if you capture `this` into a local variable there will be better minification